src/main/java/hlf/java/rest/client/util/FabricChannelUtil.java

```diff
@@ -0,0 +1,174 @@
+    signaturePolicyEnvelopeBuilder.setRule(signaturePolicyBuilder.build());
+    // get the policy
+    Policies.Policy policy =
+        Policies.Policy.newBuilder()
+            .setType(Policies.Policy.PolicyType.SIGNATURE_VALUE)
+            .setValue(signaturePolicyEnvelopeBuilder.build().toByteString())
+            .build();
+    // create config policy and return
+    return Configtx.ConfigPolicy.newBuilder()
+        .setPolicy(policy)
+        .setModPolicy(FabricClientConstants.CHANNEL_CONFIG_MOD_POLICY_ADMINS)
+        .build();
+  }
+
+  // getRolesFor returns the SignaturePolicy that has MSP
+  // with the logical conditions.
+  // For example, it is possible to design OR(msp1.member, msp2.client)
+  // this evaluates to
+  // identities: {
+  //  ... msp1
+  //  ... msp2
+  // }
+  // n out of {
+  //   n: 1
+  //   rules: {
+  //      SignaturePolicy{index: 0}
+  //      SignaturePolicy{index: 1}
+  //   }
+  // }
+  private static List<MspPrincipal.MSPPrincipal> getRolesFor(String policyFor, String orgMSPId) {
+    List<MspPrincipal.MSPPrincipal> mspPrincipals = new ArrayList<>();
+    MspPrincipal.MSPRole mspRole;
+    MspPrincipal.MSPPrincipal mspPrincipal;
+    switch (policyFor) {
+      case FabricClientConstants.CHANNEL_CONFIG_POLICY_TYPE_ADMINS:
+        mspRole =
+            MspPrincipal.MSPRole.newBuilder()
+                .setMspIdentifier(orgMSPId)
+                .setRole(MspPrincipal.MSPRole.MSPRoleType.ADMIN)
+                .build();
+        mspPrincipal =
+            MspPrincipal.MSPPrincipal.newBuilder()
+                .setPrincipal(mspRole.toByteString())
+                .setPrincipalClassification(MspPrincipal.MSPPrincipal.Classification.ROLE)
+                .build();
+        mspPrincipals.add(mspPrincipal);
+        break;
+      case FabricClientConstants.CHANNEL_CONFIG_POLICY_TYPE_WRITERS:
+        // any member who is an admin can write
+        mspRole =
+            MspPrincipal.MSPRole.newBuilder()
+                .setMspIdentifier(orgMSPId)
+                .setRole(MspPrincipal.MSPRole.MSPRoleType.ADMIN)
+                .build();
+        mspPrincipal =
+            MspPrincipal.MSPPrincipal.newBuilder()
+                .setPrincipal(mspRole.toByteString())
+                .setPrincipalClassification(MspPrincipal.MSPPrincipal.Classification.ROLE)
+                .build();
+        mspPrincipals.add(mspPrincipal);
+        // any client can also write
+        mspRole =
+            MspPrincipal.MSPRole.newBuilder()
+                .setMspIdentifier(orgMSPId)
+                .setRole(MspPrincipal.MSPRole.MSPRoleType.CLIENT)
+                .build();
+        mspPrincipal =
+            MspPrincipal.MSPPrincipal.newBuilder()
+                .setPrincipal(mspRole.toByteString())
+                .setPrincipalClassification(MspPrincipal.MSPPrincipal.Classification.ROLE)
+                .build();
+        mspPrincipals.add(mspPrincipal);
+        break;
```

Comment:
There are multiple fragments of duplicated code here. We can simplify this as

```
switch (policyFor) {
        case FabricClientConstants.CHANNEL_CONFIG_POLICY_TYPE_ADMINS:
            mspPrincipals.add(createPrincipal(orgMSPId, MspPrincipal.MSPRole.MSPRoleType.ADMIN));
            break;
        case FabricClientConstants.CHANNEL_CONFIG_POLICY_TYPE_WRITERS:
            mspPrincipals.add(createPrincipal(orgMSPId, MspPrincipal.MSPRole.MSPRoleType.ADMIN));
            mspPrincipals.add(createPrincipal(orgMSPId, MspPrincipal.MSPRole.MSPRoleType.CLIENT));
            break;
            .
            .
            // Other case statements
```
`createPrinciple` would look like

```
private static MspPrincipal.MSPPrincipal createPrincipal(String mspId, MspPrincipal.MSPRole.MSPRoleType roleType) {
    MspPrincipal.MSPRole mspRole = MspPrincipal.MSPRole.newBuilder()
        .setMspIdentifier(mspId)
        .setRole(roleType)
        .build();
    return MspPrincipal.MSPPrincipal.newBuilder()
        .setPrincipal(mspRole.toByteString())
        .setPrincipalClassification(MspPrincipal.MSPPrincipal.Classification.ROLE)
        .build();
}
```

Reply:
Good suggestion, Incorporated changes.